src/components/Datatable/Datatable.vue

```diff
@@ -0,0 +1,280 @@
+<template>
+  <table>
+    <thead class="table-head">
+      <tr v-if="windowWidth <= 1200" class="mobile-sort-container">
+        <th v-if="windowWidth <= 1200" class="mobile-sort">
+          <slot name="header-sort" :handleMobileSort="sortColumns">
+            <Button @click="sortColumns(columns[0])">Sort: A-Z</Button></slot
+          >
+        </th>
+      </tr>
+
+      <tr v-if="windowWidth >= 1200" class="table-row">
+        <th
+          v-for="(column, columnIndex) in columns"
+          class="table-header"
+          :style="{ textAlign: column.align ? column.align : 'start' }"
+          :key="columnIndex"
+          @click="sortColumns(column)"
+        >
+          <div
+            class="icon-container"
+            :style="{ textAlign: column.align ? column.align : 'start' }"
+          >
+            <slot name="column" :column="column">
+              <slot
+                :name="'column-' + columns[columnIndex].value"
+                :column="column"
+              >
+                {{ columns[columnIndex].text }}
+              </slot>
```

Comment:
We don't need to use `columnIndex` here

```suggestion
              <slot :name="`column-${column.value}`" :column="column">
                {{ column.text }}
              </slot>
```